src/type.cpp

```diff
@@ -742,6 +744,8 @@ bool TemplateTypeParmType::IsSignedType() const { return false; }
 
 bool TemplateTypeParmType::IsConstType() const { return isConst; }
 
+bool TemplateTypeParmType::IsCompleteType() const { return false; }
```

Comment:
I don't really understand how this works with templates / template paramters.

Would be good to have a test that checks that the same errors are properly reported when the same happens for function instantiation - i.e. wrt function parameters and return types.

Reply:
I have added a test to check template instantiation with an incomplete struct type.

Reply:
The tests look good, thank you!